sources/zafira-services/src/main/java/com/qaprosoft/zafira/service/integration/tool/adapter/testautomationtool/BrowserStackIntegrationAdapter.java

```
@@ -0,0 +1,86 @@
+import kong.unirest.UnirestException;
+import kong.unirest.UnirestInstance;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import org.springframework.util.StringUtils;
+
+import javax.annotation.PreDestroy;
+
+public class BrowserStackIntegrationAdapter extends AbstractIntegrationAdapter implements TestAutomationToolAdapter {
+
+    private final String url;
+    private final String username;
+    private final String accessKey;
+
+    private final UnirestInstance restClient = initClient();
+
+    public BrowserStackIntegrationAdapter(Integration integration) {
+        super(integration);
+        this.url = getAttributeValue(integration, BrowserStackParam.BROWSER_STACK_URL);
+        this.username = getAttributeValue(integration, BrowserStackParam.BROWSER_STACK_USER);
+        this.accessKey = getAttributeValue(integration, BrowserStackParam.BROWSER_STACK_ACCESS_KEY);
+    }
+
+    private UnirestInstance initClient() {
+        Config config = new Config();
+        config.connectTimeout(5000);
+        return new UnirestInstance(config);
+    }
+
+    @Getter
+    @AllArgsConstructor
+    private enum BrowserStackParam implements AdapterParam {
+        BROWSER_STACK_URL("BROWSER_STACK_URL"),
+        BROWSER_STACK_USER("BROWSER_STACK_USER"),
+        BROWSER_STACK_ACCESS_KEY("BROWSER_STACK_ACCESS_KEY");
+
+        private final String name;
+    }
+
+    @Override
+    public String buildUrl() {
+        String result = null;
+        if(!StringUtils.isEmpty(username) && !StringUtils.isEmpty(accessKey)) {
+            String[] urlSlices = url.split("//");
+            result = String.format("%s//%s:%s@%s", urlSlices[0], username, accessKey, urlSlices[1]);
+        }
+        return result != null ? result : url;
+    }
+
+    @Override
+    public boolean isConnected() {
+        try {
+            HttpResponse response = restClient.get(url).asEmpty();
+            return response.getStatus() == 200;
+        } catch (UnirestException e) {
```

Exception is just swallowed somewhere inside zafira, I think we should add logging here